store/store.go

```diff
@@ -493,6 +515,41 @@ func (s *Store[H]) get(ctx context.Context, hash header.Hash) ([]byte, error) {
 	return data, nil
 }
 
+func (s *Store[H]) addKnownHeader(h H) {
+	s.knownHeadersLk.Lock()
+	s.knownHeaders[h.Height()] = h
```

Comment:
My only concern with this map is that there's no cap on it in size. I don't think that should be a concern rn with the behaviour of syncer bc syncer forces an adjacent sync when it recognises local head is < network head, but if we have a scenario where we can't request headers for whatever reason via p2p exchange but are still receiving through gossipsub, then this map can grow forever. We actually currently have this case IIRC in `pending` as we never cap it. So not really a problem we need to urgently resolve right now but we just have to be careful with it if we do change syncer behaviour in a way which doesn't advance local head quickly enough.

Reply:
Yeah, I also thought about that and pushed the simplest fix as it is. For now we can add a new metric to see how big this queue is, but adding a limit on the map sounds not that bad.

Indeed we can set a max size for `s.knownHeaders` and stop adding newest headers but accept only oldest. WDYT ?

Reply:
Resolving 'cause we're not going to use map anymore.